code/API_definitions/EdgeCloud_LcM.yaml

```diff
@@ -0,0 +1,648 @@
+        regionId:
+          $ref: '#/components/schemas/RegionId'
+      required:
+        - artifactId
+        - appName
+    AppName:
+      type: string
+      description: Name given by the application provider
+    ArtifactId:
+      type: string
+      format: uuid
+      description: A globally unique identifier assosiated with the artifact. Originating OP generates this identifier when artifact is submitted over NBI.
+ 
+    ArtifactDetails:
+      properties:
+        name:
+          type: string
+          description: Name of the artifact. 
+        version:
+          type: integer
+          description: Artifact version information
+        virtType:
+          type: string
+          enum:
+            - VM
+            - CONTAINER
+            - HELM 
+        fileName:
+          type: string
+          minLength:  8
+          maxLength: 32
+          description: Name of the file with the extension eg. zip, targz,etc.
+        repository:       
+          type: object
+          required:
+            - type
+            - url
+          properties:
+            type:                 
+              type: string
+              enum:
+                - PRIVATEREPO
+                - PUBLICREPO
+              description: Artifact repository location. PUBLICREPO is used of public urls like github, helm repo etc, 
+                                  PRIVATEREPO  is used for private repo managed by the application developer.
+            url:
+              $ref: '#/components/schemas/Uri'
+            userName:
+              type: string
+              pattern: ^[A-Za-z][A-Za-z0-9_]{7,63}$
+              description:  Username to acces the artifact repository
+            password:
+              type: string      
+              description:  Password to acces the artifact repository
+            token:
+              type: string          
+              description:  Authorization token acces the artifact repository   
+        componentSpec:
+          description: Details about compute, networking and storage requirements for each component of the application.  
+                              App provider should  define all information needed to instantiate the component. 
+                              If artifact is being defined at component level  this section should have information just about the component. 
+                              In case the artifact is being defined at application level  the section should provide details about all the components.
+          type: object
+          items:
+            type: object
+            required:
+              - componentName
+              - operatingSystem
+              - cpuArchitecture
+              - images
+              - networkInterfaces
+              - numOfInstances
+              - restartPolicy
+            properties:
+              componentName: 
+                type: string
+                pattern:  ^[A-Za-z0-9][A-Za-z0-9_]{6,62}[A-Za-z0-9]$
```

Comment:
I think we need something like this `^[a-zA-Z]([a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?$` for RFC 1035 (stole it from [here](https://stackoverflow.com/questions/56835571/check-domain-names-according-to-rfc-1035-standard-in-java)). 
Hyphens are accepted, underscores are not, and the max is 63 (1+61+1).

Reply:
And, does it have to be a valid RFC 1035 here?

Reply:
Patterns are no longer used in the v0.2.0 for simplicity, in version 1.0.0 this specific details could be included